src/mito2/src/access_layer.rs

```diff
@@ -86,14 +96,13 @@ impl AccessLayer {
                 file_id: file_meta.file_id,
             })?;
 
-        if file_meta.inverted_index_available() {
-            let path = location::index_file_path(&self.region_dir, file_meta.file_id);
-            self.object_store
-                .delete(&path)
-                .await
-                .context(DeleteIndexSnafu {
+        let path = location::index_file_path(&self.region_dir, file_meta.file_id);
+        if let Err(err) = self.object_store.delete(&path).await {
+            if err.kind() != object_store::ErrorKind::NotFound {
+                return Err(err).context(DeleteIndexSnafu {
                     file_id: file_meta.file_id,
-                })?;
+                });
+            }
```

Comment:
If the file doesn't exist it won't return error.
https://docs.rs/opendal/latest/opendal/struct.Operator.html#method.delete

Reply:
Surprising